service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -362,49 +363,43 @@ public void createReferenceResource(ReferencedResource resource)
     storeResource(resource);
   }
 
-  @WriteTransaction
-  public boolean updateResource(
-      UUID workspaceId, UUID resourceId, String name, String description) {
-    if (name == null && description == null) {
+  private boolean updateResourceWorker(
+      UUID workspaceId,
+      UUID resourceId,
+      @Nullable String name,
+      @Nullable String description,
+      @Nullable String attributes) {
+    if (name == null && description == null && attributes == null) {
       return false;
     }
 
     var params = new MapSqlParameterSource();
 
-    if (name != null) {
+    if (!StringUtils.isEmpty(name)) {
       params.addValue("name", name);
     }
-
-    if (description != null) {
+    if (!StringUtils.isEmpty(description)) {
       params.addValue("description", description);
     }
+    if (!StringUtils.isEmpty(attributes)) {
+      params.addValue("attributes", attributes);
+    }
 
-    return updateResourceColumns(workspaceId, resourceId, params);
-  }
-
-  /**
-   * This is an open ended method for constructing the SQL update statement. To use it, build the
-   * parameter list making the param name equal to the column name you want to update. The method
-   * generates the column_name = :column_name list. It is an error if the params map is empty.
-   *
-   * @param columnParams sql parameters
-   * @param workspaceId workspace identifier - not strictly necessarily, but an extra validation
-   * @param resourceId resource identifier
-   */
-  private boolean updateResourceColumns(
-      UUID workspaceId, UUID resourceId, MapSqlParameterSource columnParams) {
     StringBuilder sb = new StringBuilder("UPDATE resource SET ");
 
-    sb.append(DbUtils.setColumnsClause(columnParams));
+    if (params.hasValue("attributes")) {
+      sb.append(DbUtils.setColumnsClause(params, "attributes"));
```

Comment:
You dont' need this conditional anymore.

Reply:
Done. also added more testings for DbUtils.